### ISSUE TITLE
Memberlist out of sync in fresh DM

### ISSUE THREAD

**Issue by lampholder** [vector-im/element-web#20962]
### Steps to reproduce

1. Received and accepted an invite to a DM
2. Chatted successfully for a bit
3. Other participant complained of UISIs
4. I noticed that Element Desktop Windows thought I was the only person in the room
5. Other Elements (Mac, Android) are fine and know that we're both in the room

#### What did you expect?
Memberlist to be accurate

#### What happened instead?
Memberlist was missing a participant

### Homeserver

me on lant.uk, other participant on their own personal homeserver (not matrix.org)

### Will you send logs?

Yes

**Comment by lampholder** [comment]
Yep, my encryption also died in this room from this device. Other devices (which maintained a correct view of participants in the room) were not affected.

**Comment by ara4n** [comment]
I just got this too. @novocaine, surely this is a release blocker?

**Comment by ara4n** [comment]
I just got this in a non-DM (my client doesn't think @Erethon is in the Full Mesh VoIP room, so he's getting UISIs from me)

**Comment by dbkr** [comment]
I've just managed to repro this (although not reliably). It's due to a member event that's pointing the wrong way: the current state of the room has ended up with a member event with `forwardLooking: false`. It has no prevContent, so the membership has been set to undefined. Current state events should always be forward looking since they're telling us about the state from this point forwards, so somehow this has got clobbered with a backwards-looking version of the event.

This plausibly could be the change we reverted (https://github.com/matrix-org/matrix-js-sdk/pull/2171) although I'm still not 100% sure how: somehow the event must be arriving from some channel that's backfilling it before it's arriving via the right source (timeline? lazy loading?).

This means I'm more confident that this is the same underlying cause as https://github.com/vector-im/element-web/issues/20981
